CodeForAfrica/wazimap-ng
# Feature: Create script to restructure shape files

## Description
Created a management command in the boundaries application to open and split shape files based on specific arguments passed in. 

### **Why**? 
The wazimap backend stores geographies in a hierarchy and this data should be representative of the loaded shape files in the application. For example, the Africa root geography should contain countries and each country should contain it's representative sub region, i.e, province, county, district etc. There are some geographies that contain multiple sub regions at the same level, e.g City, Municipality, Province can all be children of Country and this should be accurately represented in the data. This script splits the shape file into multiple files based on these children so that they can be loaded using the `loadshp` command with the correct argument, e.g `...loadshp ./cities/shape.shp Botswana City ...`

## Related Issue
#7 

## How to test it locally
Download the shapefiles from Google Drive and store them in some directory. Run the management command passing in the right options and verify that the files have been split according to your keys. e.g

`python manage.py rebuildshpfiles ./datasets/shapefiles/country-regions ENGTYPE_1,Region`


## Checklist


### Pull Request


### Commits


### Code Quality


### Testing
